pkg/framework/capi_machinesets.go

```
@@ -175,6 +176,103 @@ func WaitForCAPIMachinesRunning(ctx context.Context, cl client.Client, name stri
+			if archErr == nil {
+				return arch, nil
+			}
+
+			errs = append(errs, fmt.Errorf("error getting the architecture from CAPI worker MachineSet %s: %w", ms.Name, archErr))
+		}
+	}
+
+	if len(errs) > 0 {
+		return "", fmt.Errorf("error getting the architecture from any worker MachineSet: %w", errors.Join(errs...))
+	}
+	return "", errNoWorkerMachineSetsFound
```

golangci-lint (2.12.2)

[error] return with no blank line before

(nlreturn)